File: server/src/modules/rental-requests/domain/entities/contract-member.entity.ts
export class ContractMemberEntity {
  constructor(
    public readonly id: string,
    public readonly contractId: string,
    public readonly fullName: string,
    public readonly phone: string | null,
    public readonly identityNumber: string | null,
    public readonly email: string | null,
    public readonly address: string | null,
    public readonly isRoomLeader: boolean,
    public readonly leftAt: Date | null,
    public readonly createdAt: Date,
    public readonly updatedAt: Date,
  ) {}
}

File: client/packages/domain/lib/src/rental_request/entities/contract_member_entity.dart
abstract class ContractMemberEntity {
  ContractMemberEntity({
    required this.id,
    required this.contractId,
    required this.fullName,
    required this.isRoomLeader,
    required this.createdAt,
    required this.updatedAt,
    this.phone,
    this.identityNumber,
    this.email,
    this.address,
    this.leftAt,
  });

  final String id;
  final String contractId;
  final String fullName;
  final bool isRoomLeader;
  final String? phone;
  final String? identityNumber;
  final String? email;
  final String? address;
  final String? leftAt;
  final String createdAt;
  final String updatedAt;
}
